python/ray/node.py

```diff
@@ -411,6 +412,12 @@ def _prepare_socket_file(self, socket_path, default_prefix):
                     raise RuntimeError(
                         "Socket file {} exists!".format(socket_path))
                 try_to_create_directory(os.path.dirname(socket_path))
+
+            # Check socket path length to make sure it's short enough
+            maxlen = (104 if is_mac else 108) - 1  # sockaddr_un->sun_path
```

Comment:
we could probably add a test case in `test_tempdir.py` to illustrate this?

Reply:
Okay, I just a conservative test. I can't test the exact boundary though, since it'll be too brittle depending on the actual socket names.